src/globus_sdk/services/auth/client_types/base.py

```diff
@@ -129,37 +135,40 @@ def _convert_listarg(val):
 
         log.info("Looking up Globus Auth Identities")
 
+        if query_params is None:
+            query_params = {}
```

Comment:
To your question about ways of handling the default and guaranteeing a dict, I think this is the least-bad way to do it.
If you prefer, this is a line shorter but not in any way superior:
```python
query_params = query_params or {}
```